jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/AbstractDiagnosticsCollector.java

```diff
@@ -0,0 +1,301 @@
+/******************************************************************************* 
+ * Copyright (c) 2022 IBM Corporation and others. 
+ * 
+ * This program and the accompanying materials are made available under the 
+ * terms of the Eclipse Public License v. 2.0 which is available at 
+ * http://www.eclipse.org/legal/epl-2.0. 
+ * 
+ * SPDX-License-Identifier: EPL-2.0 
+ * 
+ * Contributors: 
+ *     IBM Corporation - initial API and implementation 
+ *******************************************************************************/ 
+package org.eclipse.lsp4jakarta.jdt.core;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.jdt.core.IAnnotation;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IImportContainer;
+import org.eclipse.jdt.core.IImportDeclaration;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IMember;
+import org.eclipse.jdt.core.ISourceRange;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.ITypeHierarchy;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.internal.core.ImportContainerInfo;
+import org.eclipse.jdt.internal.core.JavaModelManager;
+import org.eclipse.jdt.internal.corext.util.JavaModelUtil;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.eclipse.lsp4j.Range;
+
+/**
+ *
+ * Abstract class for collecting Java diagnostics.
+ *
+ */
+@SuppressWarnings("restriction")
+public abstract class AbstractDiagnosticsCollector implements DiagnosticsCollector {
+		
+    /**
+     * Constructor
+     */
+    public AbstractDiagnosticsCollector() {
+		super();

```

Comment:
Formatting still looks slightly off in this class with the indentation and mix of spaces/tabs. Maybe entire class needs to be run through the formatter